src/Servers/Kestrel/Core/src/Internal/Http3/QPack/QPackEncoder.cs

```diff
@@ -0,0 +1,504 @@
+         */
+        public static bool EncodeLiteralHeaderFieldWithPostBaseNameReference(int index, Span<byte> destination, out int bytesWritten)
+        {
+            bytesWritten = 0;
+            return false;
+        }
+
+        public static bool EncodeLiteralHeaderFieldWithoutNameReference(int index, Span<byte> destination, out int bytesWritten)
+        {
+            bytesWritten = 0;
+            return false;
+        }
+
+        /*
+         *     0   1   2   3   4   5   6   7
+               +---+---+---+---+---+---+---+---+
+               |   Required Insert Count (8+)  |
+               +---+---------------------------+
+               | S |      Delta Base (7+)      |
+               +---+---------------------------+
+               |      Compressed Headers     ...
+               +-------------------------------+
+         *
+         */
+        private static bool EncodeHeaderBlockPrefix(Span<byte> destination, out int bytesWritten)
+        {
+            int length;
+            bytesWritten = 0;
+            // Required insert count as first int
+            if (!IntegerEncoder.Encode(0, 8, destination, out length))
+            {
+                return false;
+            }
+
+            bytesWritten += length;
+            destination = destination.Slice(length);
+
+            // Delta base
+            if (destination.Length == 0)
+            {
+                return false;
+            }
+
+            destination[0] = 0x00; // zero out as we don't have a signed bit.
+            if (!IntegerEncoder.Encode(0, 7, destination, out length))
+            {
+                return false;
+            }
+
+            bytesWritten += length;
+
+            return true;
+        }
+
+        private static bool EncodeLiteralHeaderName(string value, Span<byte> destination, out int bytesWritten)
+        {
+            // From https://tools.ietf.org/html/rfc7541#section-5.2
+            // ------------------------------------------------------
+            //   0   1   2   3   4   5   6   7
+            // +---+---+---+---+---+---+---+---+
+            // | H |    String Length (7+)     |
+            // +---+---------------------------+
+            // |  String Data (Length octets)  |
+            // +-------------------------------+
+
+            if (destination.Length != 0)
+            {
+                destination[0] = 0; // TODO: Use Huffman encoding
+                if (IntegerEncoder.Encode(value.Length, 7, destination, out int integerLength))
+                {
+                    Debug.Assert(integerLength >= 1);
+
+                    destination = destination.Slice(integerLength);
+                    if (value.Length <= destination.Length)
+                    {
+                        for (int i = 0; i < value.Length; i++)
+                        {
+                            char c = value[i];
+                            destination[i] = (byte)((uint)(c - 'A') <= ('Z' - 'A') ? c | 0x20 : c);
```

Comment:
```suggestion
                            destination[i] = (byte)(c | 0x20);
```
Is this enough for header names? `[a, z]`, `[A, Z]`, `:`, `-` and numbers are handled correct.
It safes the branch. If a char is already lowercase, the `or` has no effect.

(geek note: if `value.Length` would be large enough, that could be vectorized :wink: -- but not here for header-names)

Reply:
This was copied from HPACK so I'm not going to touch it for now 😄

Reply:
👍 makes sense